app/models/institution_tree.rb

```diff
@@ -0,0 +1,86 @@
+# frozen_string_literal: true
+
+module InstitutionTree
+  def self.build(facility_code, version)
```

Comment:
following from my comment above, this would receive the institution:
```suggestion
  def self.build(institution)
    main = main_campus(institution)
```